app/main/forms.py

```diff
@@ -2591,18 +2592,142 @@ class PostcodeForm(StripWhitespaceForm):
+            DataRequired(message="Enter a radius"),
+            Only2DecimalPlaces(),
+        ],
+    )
+
+    def post_validate(self):
+        return not self.errors
+
+    def pre_validate(self, form):
+        self.first_coordinate.validate(form)
+        self.second_coordinate.validate(form)
+        return self.first_coordinate.validate(form) and self.second_coordinate.validate(form)
+
+
+class CartesianCoordinatesForm(StripWhitespaceForm):
+    """
+    This form contains the input fields for creation of areas using cartesian coordinates i.e. Eastings & northings.
+    There is additional validation post-submission of the form that checks whether the coordinate lies within
+    either the UK or the test polygons in Finland, and if both are false then an error is appended to the form.
+    """
+
+    first_coordinate = GovukDecimalField(
+        "Eastings",
+        param_extensions={
+            "classes": "govuk-input govuk-input--width-6",
+            "attributes": {"pattern": "^-?\\d+(\\.\\d+)?$"},
+        },
+        validators=[InputRequired("Enter a valid easting"), Only6DecimalPlaces()],
+    )
+    second_coordinate = GovukDecimalField(
+        "Northings",
+        param_extensions={
+            "classes": "govuk-input govuk-input--width-6",
+            "attributes": {"pattern": "^-?\\d+(\\.\\d+)?$"},
+        },
+        validators=[InputRequired("Enter a valid northing"), Only6DecimalPlaces()],
+    )
+    radius = GovukDecimalField(
+        "Radius",
+        param_extensions={
+            "classes": "govuk-input govuk-input--width-5",
+            "suffix": {"text": "km"},
+            "attributes": {"pattern": "^-?\\d+(\\.\\d+)?$"},
+        },
+        validators=[
+            NumberRange(min=0.099, max=38.001, message="Enter a radius between 0.1km and 38.0km"),
+            DataRequired(message="Enter a radius"),
             Only2DecimalPlaces(),
         ],
     )
 
     def post_validate(self):
-        return False if self.errors else True
+        return not self.errors
+
+    def pre_validate(self, form):
+        self.first_coordinate.validate(form)
+        self.second_coordinate.validate(form)
+        return self.first_coordinate.validate(form) and self.second_coordinate.validate(form)
+
+
+class ChooseCoordinateTypeForm(StripWhitespaceForm):
+    content = GovukRadiosField(
+        choices=[
+            ("decimal", "Latitude and longitude (decimal degrees)"),
+            ("cartesian", "Eastings and northings"),
+        ],
+        param_extensions={
+            "fieldset": {"legend": {"classes": "govuk-visually-hidden"}},
+            "hint": {"text": "Select one option"},
+            "items": [
+                {"hint": {"text": "For example, 51.503630, -0.126770"}},
+                {"hint": {"text": "For example, 530111, 179963"}},
+            ],
+        },
+        validators=[DataRequired(message="Select which type of coordinates you'd like to use")],
+    )
```

Comment:
I'd be tempted to use more literal values like latitude_longitude and easting_northing.  I'm not sure the text "Decimal Degrees) is helping anyone here, but perhaps our content designer has an opinion.

Reply:
I've changed all references to cartesian and decimal coordinates to instead be 'easting_northing' and 'latitude_longitude', respectively, and have checked with the team regarding the need for "decimal degrees" to be there.